Yoast/YoastSEO.js
# Relax keyphrase distribution assessment

## Summary

This PR can be summarized in the following changelog entry:

* Adjusts the scoring schema of the keyphrase distribution to return overall higher scores.

## Relevant technical choices:

* Deprecates previously introduced punishment for not using all keyphrase terms within a large window in the text
* Introduces a new punishment for having overall low scores per-sentence
* The PR also fixes the markings of the keyphrase.
* Partly duplicates PR #1865.

## Update:
* Deprecates previously introduced gini-coefficient-based estimation of the keyphrase distribution uniformity
* Instead, relies on the distance between topic usages (in sentences).
* For English and other languages with function words:
  * Short topic (<4 words): all content words should be matched for the sentence to reflect the topic.
  * Long topic ( >=4 words): more than half content words should be matched for the sentence to reflect the topic.
* For languages without function words:
  * All words should be matched for the sentence to reflect the topic.

## Test instructions

This PR can be tested by following these steps:

* Assessment is only in Premium.
* Take a few articles that rank well on Google and check if the assessment returns a Good score.
* Check that the markings work to highlight keyphrase and synonym words, not sentences without those.
* Change the keyphrase on these articles to some nonsense. The score should drop.

Fixes #1868 Fixes #1863
